Dynatrace/dynatrace-operator
# Makes image build faster for local envs

# Description

The Dockerfile was restructured in a way the any downloading we do is cached locally so we don't have to redownload the dependencies every time. (only if something changes)

~Also made the build focused `make` targets to have a `nocache` variant, if somebody wants to use that.~
~`images/push/nocache`~
~`images/push/tagged/nocache`~

Removes the cache setup step in the CI as it does not work for the current Dockerfile.
Removes the `.dockerignore` file because we only use exact paths when we use the `COPY` command. (no `COPY .` is allowed)

### DISCLAIMER
Potential question: Why didn't you do something like: (AKA.: use less build stages)
```
...
FROM go-base AS operator-build
ARG GO_LINKER_ARGS
WORKDIR /app
COPY go.mod go.sum ./
RUN go mod download && go mod verify
COPY src ./src
RUN CGO_ENABLED=1 CGO_CFLAGS="-O2 -Wno-return-local-addr" \ ...
...
```
I tried, but in that case the `RUN go mod download && go mod verify` never got cached, I don't know why, according to the docker docs it should be cached, but it just never happened. (same for the `RUN apt-get ...` stage)

## How can this be tested?
Build the image locally (with cache) 2 times, and check if it caches all the downloading parts on the 2. build.


## Checklist
